### PR TITLE
Convert all paths to the cloud.redhat.com paths

### PR DESCRIPTION
**Pull request by AllenBW**
as per https://github.com/RedHatInsights/landing-page-frontend/commit/853ec2f75702baf70591c24ca1c0858b730b9723

note, from browser console run `localStorage.setItem('cs-app', 'insights');`

### so we still have an app/nav mismatch, but @karelhala assures me, latest chrome should fix that (whenver it happens)
Screen Shot 2019-03-18 at 11 27 22 AM

### CODE REVIEW

**Comment by dkuc** [comment]
This looks right to me.  How about you @iphands

**Comment by iphands** [comment]
@AllenBW no real need to hold off until Monday.
Lemme look... But also feel free to merge away. As we break CI it might be better to test often.
Also lemme get with ya on slack and show how to test this locally.

**Comment by karelhala** [comment]
This is not complete for advisor. With this settings after opening advisor you'd see `cloud.redhat.com/apps/platform/advisor` that is not good. 
You gotta do some other changes you can follow this PR https://github.com/RedHatInsights/insights-frontend-starter-app/pull/62
* [ ] [App.js#L12 ](https://github.com/RedHatInsights/insights-advisor-frontend/blob/master/src/App.js#L12 )  `s/advisor/insights/`
* [ ] [App.js#L43](https://github.com/RedHatInsights/insights-advisor-frontend/blob/master/src/App.js#L43) no longer needed and can be removed.
* [ ] [entry.js#L14](https://github.com/RedHatInsights/insights-advisor-frontend/blob/master/src/entry.js#L14) and [entry-dev.js](https://github.com/RedHatInsights/insights-advisor-frontend/blob/master/src/entry-dev.js) should be updated

It would also be nice to add some default `spandx.config.js` so anybody can start hacking away.

**Comment by dkuc** [comment]
Thanks @karelhala !

**Comment by iphands** [comment]
Also things like https://github.com/RedHatInsights/insights-advisor-frontend/blob/master/src/Routes.js#L45
Which are not in the Starter App need to be updated.

As well as CSS that might reference assets in `/insights/platform/advisor` or `/insights/static`

File: proxy/example-local-fronend.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/*global module, process*/
+
+// Hack so that Mac OSX docker can sub in host.docker.internal instead of localhost
+// see https://docs.docker.com/docker-for-mac/networking/#i-want-to-connect-from-a-container-to-a-service-on-the-host
+const localhost = (process.env.PLATFORM === 'linux') ? 'localhost' : 'host.docker.internal';
+
+module.exports = {
+    routes: {
+        '/insights/remediations': { host: `https://${localhost}:8002` },
```
Comment (karelhala) [inline proxy/example-local-fronend.js]: `remediations` and `inventory` will be on different port because you'd run them from different webpack.

**Comment by AllenBW** [comment]
@karelhala @iphands  so this works in dev... but clicking on the lnav goes nowhere, BUT IF YOU OPEN IT IN A NEW TAB (success)🤤 🤦‍♂️... wondering if yah have any thoughts? 
Screen Shot 2019-03-18 at 10 06 03 AM

File: config/base.webpack.plugins.js [hunk 2]
```diff
@@ -100,10 +100,10 @@ const HotModuleReplacementPlugin = new webpack.HotModuleReplacementPlugin();
 plugins.push(HotModuleReplacementPlugin);
 
 /**
- * Replaces any instance of RELEASE in js files with config.insightsDeployment value.
+ * Replaces any instance of RELEASE in js files with config.deploymentEnv value.
```
Comment (karelhala) [inline config/base.webpack.plugins.js]: Should be `config.deploymentEnv` -> `config.release`

File: src/Routes.js [hunk 3]
```diff
@@ -25,26 +25,24 @@ const paths = {
...
-    return <Route { ...rest } component={ Component } />;
+    return (<Component { ...rest } />);
 };
 
 InsightsRoute.propTypes = {
     component: PropTypes.func,
     rootClass: PropTypes.string
 };
 
-function checkPaths(routes) {
```
Comment (karelhala) [inline src/Routes.js]: Do not remove this function, but change it:
```JS
function checkPaths(routes) {
return some(Object
    .values(routes), route => matchPath(location.href, { path: `${document.baseURI}insights${route}` }));
}
```

File: src/Routes.js [hunk 4]
```diff
@@ -53,23 +51,15 @@ function checkPaths(routes) {
  *      path - https://prod.foo.redhat.com:1337/insights/advisor/rules
  *      component - component to be rendered when a route has been chosen.
  */
-export const Routes = ({ childProps: { history }}) => {
-    if (!checkPaths(paths)) {
```
Comment (karelhala) [inline src/Routes.js]: You shouldn't remove this part as well, otherwise route links will break

File: src/entry-dev.js [hunk 5]
```diff
@@ -5,18 +5,14 @@ import { Provider } from 'react-redux';
 import { init } from './Store';
 import App from './App';
 import logger from 'redux-logger';
-import { NotificationsPortal } from '@red-hat-insights/insights-frontend-components/components/Notifications';
```
Comment (karelhala) [inline src/entry-dev.js]: Do not remove `NotificationsPortal` othwerwise you will loose notifications.

Comment (AllenBW) [reply src/entry-dev.js]: gonna revert, cuz no reason to do too much here

File: src/Routes.js [hunk 6]
```diff
@@ -53,23 +55,18 @@ function checkPaths(routes) {
...
+export const Routes = (props: Props) => {
+    const path = props.childProps.location.pathname;
+
     if (!checkPaths(paths)) {
         history.push(paths.actions);
     }
 
     return (
         <Switch>
-            <InsightsRoute path={ paths.actions } component={ Actions } rootClass='actions' />
```
Comment (karelhala) [inline src/Routes.js]: This shouldn't be changed because we will loose option to bookmark pages.

File: src/Routes.js [hunk 7]
```diff
@@ -53,23 +55,18 @@ function checkPaths(routes) {
  *      path - https://prod.foo.redhat.com:1337/insights/advisor/rules
  *      component - component to be rendered when a route has been chosen.
  */
-export const Routes = ({ childProps: { history }}) => {
+export const Routes = (props: Props) => {
```
Comment (karelhala) [inline src/Routes.js]: This shouldn't change, because we are handling redirect in `checkPaths` function.

File: src/Routes.js [hunk 8]
```diff
@@ -29,20 +29,22 @@ const InsightsRoute = ({ component: Component, rootClass, ...rest }) => {
...
 InsightsRoute.propTypes = {
     component: PropTypes.func,
     rootClass: PropTypes.string
 };
 
-function checkPaths(routes) {
+function checkPaths (routes) {
     return some(Object
-    .values(routes), route => matchPath(location.href, { path: `${document.baseURI}platform/advisor${route}` }));
+    .values(routes), route => matchPath(location.href, { path: `${document.baseURI}insights/advisor${route}` }));
```
Comment (karelhala) [inline src/Routes.js]: This should be just `insights` no `advisor`.

`.values(routes), route => matchPath(location.href, { path: `${document.baseURI}insights${route}` }));`

File: src/Routes.js [hunk 9]
```diff
@@ -29,20 +29,22 @@ const InsightsRoute = ({ component: Component, rootClass, ...rest }) => {
     const root = document.getElementById('root');
     root.removeAttribute('class');
     root.classList.add(`page__${rootClass}`, 'pf-c-page__main');
-    root.classList.add(`page__${rootClass}`, 'pf-l-page__main');
     root.setAttribute('role', 'main');
 
-    return <Route { ...rest } component={ Component } />;
```
Comment (karelhala) [inline src/Routes.js]: Do not change this otherwise you'd get incorrect path handling.

File: src/App.js [hunk 10]
```diff
@@ -10,13 +9,6 @@ class App extends Component {
     componentDidMount () {
         insights.chrome.init();
         insights.chrome.identifyApp('advisor');
-        insights.chrome.navigation(buildNavigation());
-        this.appNav = insights.chrome.on('APP_NAVIGATION', event => {
```
Comment (karelhala) [inline src/App.js]: This shouldn't be removed but updated
```JS
this.appNav = insights.chrome.on('APP_NAVIGATION', event => {
            if (!matchPath(location.href, { path: `${document.baseURI}insights/${event.navId}` })) {
                this.props.history.push(`/${event.navId}`);
            }
        });
```

**Pull request merged**